optaplanner-core/src/main/java/org/optaplanner/core/impl/exhaustivesearch/DefaultExhaustiveSearchPhase.java

```diff
@@ -210,14 +209,11 @@ protected void restoreWorkingSolution(ExhaustiveSearchStepScope<Solution_> stepS
                 oldNode = oldNode.getParent();
             }
         }
-        List<Move> restoreMoveList = new ArrayList<>(oldMoveList.size() + newMoveList.size());
+        List<Move<Solution_>> restoreMoveList = new ArrayList<>(oldMoveList.size() + newMoveList.size());
         restoreMoveList.addAll(oldMoveList);
         Collections.reverse(newMoveList);
         restoreMoveList.addAll(newMoveList);
-        ScoreDirector scoreDirector = phaseScope.getScoreDirector();
-        for (Move restoreMove : restoreMoveList) {
-            restoreMove.doMove(scoreDirector);
-        }
+        restoreMoveList.forEach(restoreMove -> restoreMove.doMove(phaseScope.getScoreDirector()));
```

Comment:
Nitpick: this actually does the getScoreDiretor() call n times, instead of just once. Not that matters much because it's a simple getter that is probably inlined, but Cliff Click (the JVM hotspot compiler guy) argues that inlining is not always guaranteed.

I 'll merge as is anyway.